bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/internal/ImageList.java

```diff
@@ -334,16 +334,23 @@ public long getHandle(int targetZoom) {
 	if (!zoomToHandle.containsKey(targetZoom)) {
 		int scaledWidth = DPIUtil.scaleUp(DPIUtil.scaleDown(width, this.zoom), targetZoom);
 		int scaledHeight = DPIUtil.scaleUp(DPIUtil.scaleDown(height, this.zoom), targetZoom);
-		long handle = OS.ImageList_Create(scaledWidth, scaledHeight, flags, 16, 16);
-		int count = OS.ImageList_GetImageCount(handle);
-		for (int i = 0; i < images.length; i++) {
+		long newImageListHandle = OS.ImageList_Create(scaledWidth, scaledHeight, flags, 16, 16);
+		int count = OS.ImageList_GetImageCount (handle);
+		for (int i = 0; i < count; i++) {
 			Image image = images[i];
 			if (image != null) {
-				set(i, image, count, handle, targetZoom);
-				count++;
+				set(i, image, i, newImageListHandle, targetZoom);
+			} else {
+				long hDC = OS.GetDC (0);
+				if (hDC == 0) SWT.error(SWT.ERROR_NO_HANDLES);
+				long placeholderBitmapHandle = OS.CreateCompatibleBitmap(hDC, scaledWidth, scaledHeight);
+				if (placeholderBitmapHandle == 0) SWT.error(SWT.ERROR_NO_HANDLES);
+				OS.ImageList_Add(newImageListHandle, placeholderBitmapHandle, placeholderBitmapHandle);
+				OS.DeleteObject(placeholderBitmapHandle);
+				OS.ReleaseDC(0, hDC);
```

Comment:
Would be good to extract this into a method with a self-describing name.

Reply:
I extracted it into a separate method